File: hoo_frontend/src/common/types/motion.ts
export class HooMotionSensor {
    public readonly name: string;
    public state: MotionState;

    constructor(name: string, state: MotionState) {
        this.name = name;
        this.state = state;
    }
}

export interface MotionState {
    presence: boolean;
}

File: hoo_api/src/motion.rs
use serde::{Serialize, Deserialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Motion {
    pub name: String,
    pub state: MotionState,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct MotionState {
    pub presence: bool,
}
